add context to client data fetching page aiming to be easier to understand (#15020)
gatsbyjs/gatsby: docs/docs/client-data-fetching.md

title: Client data fetching
---

## Context

> In this article it is touched **how** to fetch data and **how** to handle that data using the plugin [`gatsby-source-graphql`](/packages/gatsby-source-graphql/).

Reviewer: ```suggestion
This article touches on _how to fetch_ data and _how to handle_ that data using the plugin [`gatsby-source-graphql`](/packages/gatsby-source-graphql/).
```

> When this article says "hydration" it means that Gatsby through ReactJS already know the markup of your site because it was rendered in the server, so React will preserve the markup and attach event handlers to it, the result of this process allows you to have a really nice performant first-load time experience.

Reviewer: ```suggestion
When this article mentions [hydration](/docs/glossary#hydration), it means that Gatsby (through React.js) builds static files to render server-side. When a page loads and React's [client-side](/docs/glossary#client-side) code downloads and executes, it preserves the HTML markup built by Gatsby and turns the site into a full JavaScript web application that can manipulate the [DOM](/docs/glossary#dom). The result of this process creates fast loading pages and a nice user experience.
```

> Build-time is useful when we know that the markup of our websites will not change or we do not really care if it changes, as mentioned before, the end-result of this process is a really fast first-load time experience as for run-time is appropriate for constantly changing markup like a chat or an email client web app.

Reviewer: ```suggestion
Compiling pages at [build-time](/docs/glossary#build) is useful when your website content will not change often, or triggering a build process to recompile is fine. However, some websites have more dynamic needs requiring [client-side](/docs/glossary#client-side) code to handle constantly changing content, like a chat widget or an email client web application.
```

> We also talk about build-time which is defined for the process when Gatsby loads all the configuration and set-up everything to start up the website and run-time which is defined for the process when Gatsby has loaded everything up and the website is running in the server environment.

Reviewer: I think this paragraph could be removed, as it seems more likely to confuse than clarify things.
